app/core/utils.py

```diff
@@ -174,6 +175,27 @@ def pil_image_to_bytes(img: Image.Image, format: str = "JPEG") -> bytes:
         return buffer.getvalue()
 
 
+class TimestampedTTLCache(cachetools.TTLCache):
+    """TTLCache subclass that tracks when items were added to the cache."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.__timestamps = {}  # Store timestamps for each key
```

Comment:
why the double-underscore ("name mangling") here? https://realpython.com/python-double-underscore/#double-leading-underscore-in-classes-pythons-name-mangling

Reply:
I used Cursor to help auto-generate this and the classes that this inherits from makes use of the double underscores, so I left it in to keep it consistent with those cachetools classes. But reading this it seems like it doesn't provide any value for the timestamps attribute, so I'll remove it.